tapdb/multiverse_cache.go

```diff
@@ -67,80 +65,86 @@ func DefaultMultiverseCacheConfig() MultiverseCacheConfig {
 	}
 }
 
-// ProofKey is used to uniquely identify a proof within a universe. This is
-// used for the LRU cache for the proofs themselves, which are considered to be
-// immutable.
-type ProofKey [32]byte
-
-// NewProofKey takes a universe identifier and leaf key, and returns a proof
-// key.
-func NewProofKey(id universe.Identifier, key universe.LeafKey) ProofKey {
-	idBytes := id.Bytes()
-	leafKeyBytes := key.UniverseKey()
-
-	// The proof key maps down the ID and the leaf key into a single
-	// 32-byte value: sha256(id || leaf_key)..
-	h := sha256.New()
-	h.Write(idBytes[:])
-	h.Write(leafKeyBytes[:])
-
-	return fn.ToArray[ProofKey](h.Sum(nil))
-}
-
 // cachedProofs is a list of cached proof leaves.
 type cachedProofs []*universe.Proof
 
-// Size just returns 1 as we're limiting based on the total number of different
-// leaf keys we query by. So we might store more than one proof per cache entry
-// if the universe key's script key isn't set. But we only want a certain number
-// of different keys stored in the cache.
+// Size returns the total byte size of all cached proofs.
 func (c *cachedProofs) Size() (uint64, error) {
-	return 1, nil
+	if c == nil {
+		return 0, nil
+	}
+
+	totalBytes := uint64(0)
+	for _, proof := range *c {
+		if proof == nil {
+			continue
+		}
+
+		totalBytes += proof.LowerBoundByteSize()
```

Comment:
Would be nice having a caching layer for this cache, in order to avoid size calculating traversals over entries that have had their size calculated already
`entrySizeCache map[UniverseProofKey] uint64`